test/modules/standard/gmp/ferguson/gmp_dist_array.chpl

```diff
@@ -44,16 +41,27 @@ modulus.nextprime(modulus);
 writeln(modulus);
 sum.mod(sum, modulus);
 writeln(sum);
-delete modulus;
-delete sum;
+
 
 // Release the array elements.
+// TODO: remove this section once the records are being released correctly.
```

Comment:
the distributed array leaks all of its array elements. Since the elements are leaked, the records don't go out of scope and their destructors aren't called. Manually calling the destructors significantly reduces the amount of memory leaked. 

Ideally, it wouldn't need to think about the memory management, and ideally it wouldn't be leaking the array elements. In the meantime, manually freeing the records means that we leak around 2000x less data from this one test.